judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/AutoMappedTransferObjectSingleEntityTest.java

```diff
@@ -0,0 +1,206 @@
+                autoMappedContainerSingleCompositionDao.create(AutoMappedContainerSingleComposition
+                        .builder()
+                        .withSingleCompositionEntity(autoMappedReferenceEntityDao.create(AutoMappedReferenceEntity.builder().withName("singleComposition").build()))
+                        .withSingleRequiredCompositionEntity(autoMappedReferenceEntityDao.create(AutoMappedReferenceEntity.builder().withName("singleRequiredComposition").build()))
+                        .build()
+                );
+
+        // TODO JNG-4909 The field not contains the default value
+//        assertEquals(autoMappedReferenceEntityRequiredCompositionDefault.identifier(), autoMappedContainerSingleComposition.getSingleCompositionDefaultEntity().orElseThrow().identifier());
+//        autoMappedContainerSingleComposition = autoMappedContainerSingleCompositionDao.update(autoMappedContainerSingleComposition);
+//        assertEquals(autoMappedReferenceEntityRequiredCompositionDefault.identifier(), autoMappedContainerSingleComposition.getSingleCompositionDefaultEntity().orElseThrow().identifier());
+
+        assertEquals(Optional.of("singleComposition"), autoMappedContainerSingleComposition.getSingleCompositionEntity().orElseThrow().getName());
+        assertEquals(Optional.of("singleRequiredComposition"), autoMappedContainerSingleComposition.getSingleRequiredCompositionEntity().getName());
+
+        // update the name of the compositions
+
+        autoMappedContainerSingleComposition.getSingleCompositionEntity().orElseThrow().setName("singleCompositionRenamed");
+        autoMappedContainerSingleComposition.getSingleRequiredCompositionEntity().setName("singleRequiredCompositionRenamed");
+
+        autoMappedContainerSingleComposition = autoMappedContainerSingleCompositionDao.update(autoMappedContainerSingleComposition);
+
+        assertEquals(Optional.of("singleCompositionRenamed"), autoMappedContainerSingleComposition.getSingleCompositionEntity().orElseThrow().getName());
+        assertEquals(Optional.of("singleRequiredCompositionRenamed"), autoMappedContainerSingleComposition.getSingleRequiredCompositionEntity().getName());
+
+        // composition deletion
+
+        autoMappedContainerSingleComposition.setSingleCompositionEntity(null);
+        autoMappedContainerSingleComposition.setSingleCompositionEntity(autoMappedReferenceEntityDao.create(AutoMappedReferenceEntity.builder().withName("singleCompositionBind").build()));
+
+        AutoMappedContainerSingleComposition referenceForLambda = autoMappedContainerSingleComposition;
+        IllegalStateException thrown = assertThrows(
+                IllegalStateException.class,
+                () -> autoMappedContainerSingleCompositionDao.update(referenceForLambda)
+        );
+
+        assertTrue(thrown.getMessage().contains("Identifier cannot be different on containment reference element"));
+        assertTrue(thrown.getMessage().contains("#singleCompositionEntity"));
```

Comment:
Please also check this case with `autoMappedReferenceEntityDao.removeSingleCompositionEntity` and check if related element is deleted as well.

Reply:
Dao doesnt contains remove method because it is an aggregation

Reply:
[8a60227](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/195/commits/8a6022773adf349a2c4ae1c1d81c3251da09ec84)